From c62a108c79db49cecb9f323d6567d33a86fa6dc7 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Mon, 18 Mar 2019 13:54:56 +0000
Subject: [PATCH] 02_ignition_firstboot: Enable networking if Ignition will run

Today coreos-assembler is hardcoding the initramfs networking kargs to
do DHCP; this is problematic for e.g. bare metal where one often wants
to configure static networking.

Let's only enable initramfs networking if we're doing Ignition.
---
 grub/02_ignition_firstboot | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub/02_ignition_firstboot b/grub/02_ignition_firstboot
index a154918..b4e926d 100755
--- a/grub/02_ignition_firstboot
+++ b/grub/02_ignition_firstboot
@@ -9,5 +9,5 @@ search --set=bootpart --label boot
 # to be used later on the kernel command line.
 set ignition_firstboot=""
 if [ -f "(${bootpart})/ignition.firstboot" ]; then
-    set ignition_firstboot="ignition.firstboot"
+    set ignition_firstboot="ignition.firstboot rd.neednet=1 ip=dhcp"
 fi
-- 
2.20.1

